dist_run.py

```diff
@@ -84,12 +84,19 @@ def main():
     nstages = pp_mesh.size()
     device = torch.device(f"cuda:{rank}")
 
+    # Fill in PP configs
+    config.stage_idx = pp_rank
+    config.n_stages = nstages
+
     with device:
-        with tp_mesh:
-            model = TransformerStage(config, pp_rank, nstages)
-            model.setup_caches(1, 4096)
-            # TODO: refine this .to once we start using fp8 for KV cache
-            model = model.to(model_dtype)
+        model = Transformer(config)
+
+    model.setup_caches(1, 4096)
+    # TODO: refine this .to once we start using fp8 for KV cache
```

Comment:
If KV cache is the only consideration in type conversion, you might just invert the order?